dd-java-agent/appsec/src/main/java/com/datadog/appsec/config/AppSecConfigServiceImpl.java

```diff
@@ -601,4 +608,59 @@ private static WafConfig createWafConfig(Config config) {
     }
     return wafConfig;
   }
+
+  private static class SafeMapAdapter extends JsonAdapter<Object> {
+    @Override
+    public Object fromJson(JsonReader reader) throws IOException {
+      switch (reader.peek()) {
+        case BEGIN_OBJECT:
+          Map<String, Object> map = new LinkedHashMap<>();
+          reader.beginObject();
+          while (reader.hasNext()) {
+            map.put(reader.nextName(), fromJson(reader));
+          }
+          reader.endObject();
+          return map;
+
+        case BEGIN_ARRAY:
+          List<Object> list = new ArrayList<>();
+          reader.beginArray();
+          while (reader.hasNext()) {
+            list.add(fromJson(reader));
+          }
+          reader.endArray();
+          return list;
+
+        case STRING:
+          return reader.nextString();
+        case NUMBER:
```

Comment:
Minor style (added new line, to be consistent with other cases):
```suggestion
        case STRING:
          return reader.nextString();

        case NUMBER:
```